dev/CommandBarFlyout/CommandBarFlyoutCommandBar.cpp

```diff
@@ -475,6 +659,433 @@ void CommandBarFlyoutCommandBar::UpdateTemplateSettings()
+{
+    for (const auto& command : commands)
+    {
+        if (auto commandAsControl = command.try_as<winrt::Control>())
+        {
+            if (commandAsControl.IsTabStop())
+            {
+                return commandAsControl;
+            }
+        }
+    }
+    return nullptr;
+}
+
+bool CommandBarFlyoutCommandBar::FocusControl(
+    winrt::Control const& newFocus,
+    winrt::Control const& oldFocus,
+    winrt::FocusState const& focusState,
+    bool updateTabStop)
+{
+    MUX_ASSERT(newFocus);
+
+    if (updateTabStop)
+    {
+        newFocus.IsTabStop(true);
+    }
+
+    if (newFocus.Focus(focusState))
+    {
+        if (oldFocus && updateTabStop)
+        {
+            oldFocus.IsTabStop(false);
+        }
+        return true;
+    }
+    return false;
+}
+
+bool CommandBarFlyoutCommandBar::FocusCommand(
+    winrt::IObservableVector<winrt::ICommandBarElement> const& commands,
+    winrt::Control const& moreButton,
+    winrt::FocusState const& focusState,
+    bool firstCommand,
+    bool ensureTabStopUnicity)
+{
+    COMMANDBARFLYOUT_TRACE_VERBOSE(nullptr, TRACE_MSG_METH, METH_NAME, nullptr);
+
+    MUX_ASSERT(commands);
+
+    // Give focus to the first or last focusable command
+    winrt::Control focusedControl = nullptr;
+    int startIndex = 0;
+    int endIndex = static_cast<int>(commands.Size());
+    int deltaIndex = 1;
+
+    if (!firstCommand)
+    {
+        deltaIndex = -1;
+        startIndex = endIndex - 1;
+        endIndex = -1;
+    }
+
+    for (int index = startIndex; index != endIndex; index += deltaIndex)
+    {
+        auto command = commands.GetAt(index);
+
+        if (auto commandAsControl = command.try_as<winrt::Control>())
+        {
+            if (IsControlFocusable(commandAsControl, !ensureTabStopUnicity /*checkTabStop*/))
+            {
+                if (!focusedControl)
+                {
+                    if (FocusControl(
+                            commandAsControl /*newFocus*/,
+                            nullptr /*oldFocus*/,
+                            focusState /*focusState*/,
+                            ensureTabStopUnicity /*updateTabStop*/))
+                    {
+                        if (ensureTabStopUnicity && moreButton && moreButton.IsTabStop())
+                        {
+                            moreButton.IsTabStop(false);
+                        }
+
+                        focusedControl = commandAsControl;
+
+                        if (!ensureTabStopUnicity)
+                        {
+                            break;
+                        }
+                    }
+                }
+                else if (focusedControl && commandAsControl.IsTabStop())
+                {
+                    commandAsControl.IsTabStop(false);
+                }
+            }
+        }
+    }
+
+    return focusedControl != nullptr;
+}
+
+void CommandBarFlyoutCommandBar::EnsureTabStopUnicity(
```

Comment:
I'd never even heard of the word "unicity" before, and only some dictionaries seem to list it.  Perhaps "uniqueness" would be more easily understood?

Reply:
Changed to uniqueness.